custom_components/zwave_mqtt/__init__.py

```diff
@@ -246,6 +250,35 @@ async def handle_remove_node(hass: HomeAssistant, node_id: int):
             dev_registry.async_remove_device(dev_id)
 
 
+async def handle_node_update(hass: HomeAssistant, node: OZWNode):
+    """
+    Handle a node updated event from OZW.
+
+    Meaning some of the basic info like name/model is updated.
+    We want these changes to be pushed to the device registry.
+    """
+    dev_registry = await get_dev_reg(hass)
+    # grab device in device registry attached to this node
+    device = dev_registry.async_get_device([(DOMAIN, node.id)], [])
+    if device:
+        # update device in device registry with (updated) info
+        # also handle any slave devices (node instances)
+        for item in dev_registry.devices.values():
+            if item.id == device.id or item.via_device_id == device.id:
+                dev_name = f"{node.node_manufacturer_name} {node.node_product_name}"
```

Comment:
Home Assistant already stores manufacturer and product. Seems weird to put in device name too?

Reply:
This is used as the default name the device gets but you're right, this does not have to be the full devicename and manufacturer. We can also choose some shorter default name like "Z-Wave Node 1"

Reply:
I'm working on another PR that has focus on device/entity names to make things more user friendly. I will take this naming improvement there. So leave it here for now to narrow down scope.